app/services/image_blob_service.rb

```diff
@@ -31,9 +32,11 @@ def call
 
 private
 
-  attr_reader :revision, :user, :temp_image
+  attr_reader :temp_image, :revision, :user, :hardcoded_filename
 
   def unique_filename
+    return hardcoded_filename if hardcoded_filename.present?
```

Comment:
We still need this filename to be unique so we should run this through the unique filename service

Reply:
Have been thinking about this, and not sure it's a good idea to run through the UniqueFilenameService, as this can end up changing the filename if duplicates are detected. And then what? The Govspeak body would be out of date, referencing images by filenames that no longer exist.

For the same reason, I don't think we should parameterize the filenames (e.g. `Foo--1.png` => `foo-1.png`), which happens when we call the UniqueFilenameService regardless of the uniqueness of the filename.

I'd be inclined to check for duplicate image filenames manually and abort the import when that happens. This leaves me with two concerns:

1) What happens if an editor attempts to edit an image that originated from Whitehall? (Presumably if there is buggy behaviour they could work around it by deleting the image and reuploading).
2) Are we in any way tied to the `/media/hash-id/filename.extension` structure (as opposed to `/government/uploads/system/uploads/image_data/file/file-id/filename.extension`)? The [`asset_manager_id` method](https://github.com/alphagov/content-publisher/blob/5acbe8293917ea0b90a96c23b8768e708ebbd2b9/app/models/image/asset.rb#L23-L27)) suggests we might be, but as the portion of the URL preceding the filename acts as an identifier in either case, it may not be an issue.